Fix unclosed rocks iterators

xandkar opened the pull request.

jeffgrunewald commented on src/blockchain.erl:
```diff
@@ -892,12 +892,18 @@ find_first_height_after(MinHeight0, #blockchain{db=DB, heights=HeightsCF}) ->
+                lager:error("Unexpected iterator error: ~p", [Reason]),
+                {error, not_found}
+        end,
+    catch rocksdb:iterator_close(Iter),
```
thought we were trying to avoid these types of catches since they swallowed errors

xandkar replied:
Yeah. I hate it. I've done a bunch of rocks iteration experiments recently and stumbled on enough seemingly-nondeterministic iterator-access errors to make me nervous.

While I lean on removing these `catch`s here, I'm not sure what will happen if you try to close an iterator after an error other than `invalid_iterator` - hence the recon logging.

I wasn't able to reproduce those not-`invalid_iterator` errors.

xandkar replied:
@jeffgrunewald OK, I replaced all the `catch rocksdb:iterator_close`s with `try rocksdb:iterator_close catch log end`s.

jeffgrunewald reviewed (approved):
Love the macro!

evanmcc commented on src/blockchain_rocks.hrl:
```diff
@@ -0,0 +1,10 @@
+        try
+            rocksdb:iterator_close(IteratorHandle)
+        catch ErrorClass:ErrorReason ->
+            lager:error("rocksdb:iterator_close error: ~p:~p", [ErrorClass, ErrorReason])
```
change to warning, please

xandkar replied:
done.

evanmcc commented:
suspect that the catch logging is going to be noisy, we might want to add a version of the macro that allows us not to log.

in any case, we want to log at `warning`, since error writes to the error log with `fsync`

xandkar commented:
> suspect that the catch logging is going to be noisy, we might want to add a version of the macro that allows us not to log.

added the an an app env var to disable the logging.

> in any case, we want to log at `warning`

switched it to warning.

> since error writes to the error log with `fsync`

TIL!
